Enable CI

sveitser opened the pull request:
For reviewing check my commits, the rest are synced from upstream.

Close https://github.com/EspressoSystems/espresso-sequencer/issues/689

amsanghi committed:
Test that parsing an empty CLI option set results in a default config

amsanghi committed:
Use update geth for missing config

amsanghi committed:
update gitmodules

amsanghi committed:
update gitmodules

amsanghi committed:
update geth

amsanghi committed:
update gitmodules

amsanghi committed:
update gitmodules

amsanghi committed:
Fix test

amsanghi committed:
update geth

amsanghi committed:
update geth

amsanghi committed:
Merge branch 'master' into test_empty_cli

amsanghi committed:
Empty-Commit

amsanghi committed:
Changes based on offline discussion

amsanghi committed:
Merge branch 'master' into test_empty_cli

amsanghi committed:
minor fix

ganeshvanahalli committed:
initial test node builder pattern and few impl

ganeshvanahalli committed:
refactor more impl

ganeshvanahalli committed:
resolve merge conflicts

ganeshvanahalli committed:
Merge branch 'master' into test-node-builder

ganeshvanahalli committed:
update implementation and limit diff to a few testfiles for avoiding merge conflicts

amsanghi committed:
Merge branch 'master' into test_empty_cli

amsanghi committed:
minor fix

amsanghi committed:
minor fix

amsanghi committed:
minor fix

amsanghi committed:
minor fix

ganeshvanahalli committed:
Update builder pattern

ganeshvanahalli committed:
fix typo

ganeshvanahalli committed:
address PR comments

ganeshvanahalli committed:
handle defaults

ganeshvanahalli committed:
address PR comments

ganeshvanahalli committed:
remove buildL1/L2 blockchain fns

ganeshvanahalli committed:
resolve merge conflicts

ganeshvanahalli committed:
refactor Build2ndNode

anodar committed:
Add access lists for batchposter transactions to reduce cold memory access cost

ganeshvanahalli committed:
add ExecutionNode to TestClient

anodar committed:
Merge branch 'master' into access-lists

anodar committed:
Generate dynamic slots, add tests

anodar committed:
drop gas-refunder from arbitrum_chain_info.json

anodar committed:
Move sequencer ABI instantiation outside loop

anodar committed:
Drop commented out line from StorageSlotAddress

anodar committed:
Merge branch 'master' into access-lists

anodar committed:
Add access list entries for dynamic storage slots for bridge contract

anodar committed:
drop remnant from debugging

anodar committed:
Merge branch 'access-lists' into access-lists-cont

anodar committed:
Merge pull request #1905 from OffchainLabs/access-lists

Create access lists for batchposter transactions to reduce memory access cost

anodar committed:
Merge branch 'master' into access-lists-cont

amsanghi committed:
Merge branch 'master' into test_empty_cli

anodar committed:
Use big.Int for bytes sum and int to byte conversion

anodar committed:
Merge branch 'access-lists-cont' of github.com:OffchainLabs/nitro into access-lists-cont

amsanghi committed:
minor fix

anodar committed:
Merge pull request #1907 from OffchainLabs/access-lists-cont

Add access list entries for dynamic storage slots for bridge contract

joshuacolvin0 committed:
Add new metrics for inbox

The following metrics are added to track the state currently read from inbox:
* arb_inbox_batch
* arb_inbox_message

potuz committed:
Fix typo and let potuz have a nitro PR

joshuacolvin0 committed:
Merge pull request #1910 from potuz/rorg_not_reorg

Fix typo and let potuz have a nitro PR

joshuacolvin0 committed:
Merge branch 'master' into inbox-metrics

joshuacolvin0 committed:
Update metric names

* arb_inbox_latest_batch
* arb_inbox_latest_batch_message

amsanghi committed:
Merge branch 'master' into test_empty_cli

amsanghi committed:
Merge pull request #1661 from OffchainLabs/test_empty_cli

Test that parsing an empty CLI option set results in a default config

anodar committed:
Use options struct for batchposter to make callsites more readable

GoodDaisy committed:
fix typos

magicxyyz committed:
add triedb race test

magicxyyz committed:
improve test configs in recreatestate_rpc_test

magicxyyz committed:
Merge branch 'master' into triedb-race-test

magicxyyz committed:
fix triedb_race_test

magicxyyz committed:
set ParentChainReader default config in gethexec.ConfigDefaultTest

magicxyyz committed:
update geth to version with triedb race fix

magicxyyz committed:
update geth

magicxyyz committed:
update geth

ganeshvanahalli committed:
Add --dev flag to easily start up a local L2-only dev chain

magicxyyz committed:
update geth

anodar committed:
Merge pull request #1911 from OffchainLabs/cleanup

Use options struct for batchposter to make callsites more readable

ganeshvanahalli committed:
address PR comments and resolve conflicts

ganeshvanahalli committed:
Add a CI step to start up a dev test node

PlasmaPower committed:
Don't ignore target/machines for Docker builds

ganeshvanahalli committed:
fix issue

ganeshvanahalli committed:
use run-nitro-test-node action impl

PlasmaPower committed:
Merge branch 'master' into docker-dont-ignore-machines

PlasmaPower committed:
Merge pull request #1918 from OffchainLabs/docker-dont-ignore-machines

Don't ignore target/machines for Docker builds

ganeshvanahalli committed:
change image name

ganeshvanahalli committed:
update nitro-testnode pin

GoodDaisy committed:
Merge branch 'master' into master

ganeshvanahalli committed:
init only

anodar committed:
Implement external signer for dataposter

anodar committed:
Merge branch 'master' into external-signer

anodar committed:
merge master

anodar committed:
Drop accidentally added server binary for manual testing

anodar committed:
Fix linter errors

ganeshvanahalli committed:
Merge branch 'master' into nitro-default-devchain

anodar committed:
Merge branch 'master' into external-signer

ganeshvanahalli committed:
run int dev in foreground

anodar committed:
Make signing method configurable to allow both eth_signTransaction and account_signTransaction

anodar committed:
Merge branch 'external-signer' of github.com:OffchainLabs/nitro into external-signer

ganeshvanahalli committed:
Merge pull request #1914 from OffchainLabs/nitro-default-devchain

Add --dev flag to easily start up a local L2-only dev chain

joshuacolvin0 committed:
Merge branch 'master' into inbox-metrics

joshuacolvin0 committed:
Merge pull request #1909 from OffchainLabs/inbox-metrics

Add new metrics for inbox

ganeshvanahalli committed:
update testnode-pin, fix bugs, test run nitro-testnode on foreground

joshuacolvin0 committed:
Merge branch 'master' into test-node-builder

ganeshvanahalli committed:
run testnode in background

joshuacolvin0 committed:
Merge pull request #1883 from OffchainLabs/test-node-builder

Node builder pattern for running test nodes in tests

joshuacolvin0 committed:
Change L1 to parent chain in DAS help strings

joshuacolvin0 committed:
Merge branch 'master' into add-testnodestart-ci

joshuacolvin0 committed:
Merge pull request #1917 from OffchainLabs/add-testnodestart-ci

Add a CI step to start up a dev test node

joshuacolvin0 committed:
Merge branch 'master' into master

joshuacolvin0 committed:
Make clippy happy

joshuacolvin0 committed:
Make cargo fmt happy

PlasmaPower committed:
Merge pull request #1921 from OffchainLabs/clippy-fix

Make clippy happy

joshuacolvin0 committed:
Merge branch 'master' into master

Tristan-Wilson committed:
Merge branch 'master' into das-help-strings

Tristan-Wilson committed:
Merge branch 'master' into das-help-strings

Tristan-Wilson committed:
Merge pull request #1920 from OffchainLabs/das-help-strings

joshuacolvin0 committed:
Merge branch 'master' into master

joshuacolvin0 committed:
Merge pull request #1912 from GoodDaisy/master

fix typos

anodar committed:
Update arbnode/dataposter/data_poster.go

Co-authored-by: Joshua Colvin <jcolvin@offchainlabs.com>

anodar committed:
Merge branch 'master' into external-signer

anodar committed:
Check that signed transaction is the one sent for signing

anodar committed:
Merge branch 'external-signer' of github.com:OffchainLabs/nitro into external-signer

anodar committed:
List transactions in correct order in the log line

anodar committed:
Add TLS to the server side, expose optional root certificate path on client side

anodar committed:
Set min tls verion

Tristan-Wilson committed:
Add --log-type option for daserver

This adds the ability to have json logs.

Tristan-Wilson committed:
Merge pull request #1925 from OffchainLabs/json-log-type-daserver

Add --log-type option for daserver

PlasmaPower committed:
Change default sequencer MaxBlockSpeed to 250ms

PlasmaPower committed:
Merge branch 'master' into default-250ms-max-block-speed

PlasmaPower committed:
Merge pull request #1927 from OffchainLabs/default-250ms-max-block-speed

Change default sequencer MaxBlockSpeed to 250ms

anodar committed:
Merge branch 'master' into external-signer

anodar committed:
Add default method for external signer

anodar committed:
Merge branch 'external-signer' of github.com:OffchainLabs/nitro into external-signer

anodar committed:
Change port 443 back to 1234

anodar committed:
Merge branch 'external-signer' into external-signer-mtls

anodar committed:
Merge pull request #1919 from OffchainLabs/external-signer

Implement external signer for dataposter

PlasmaPower committed:
Merge pull request #1923 from OffchainLabs/external-signer-mtls

Add TLS to the server side, expose optional root certificate path on client side

sveitser committed:
ci: use ubuntu-latest

joshuacolvin0 committed:
Merge branch 'master' into triedb-race-test

joshuacolvin0 committed:
Merge pull request #1913 from OffchainLabs/triedb-race-test

Add triedb race test

sveitser committed:
Merge remote-tracking branch 'upstream/master' into ma/ci

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EspressoSystems/nitro-espresso-integration?pullRequest=2) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/nitro-espresso-integration?pullRequest=2) before we can accept your contribution.<br/>**1** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: sveitser<br/>:x: PlasmaPower<br/>:x: joshuacolvin0<br/>:x: Tristan-Wilson<br/>:x: anodar<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/nitro-espresso-integration?pullRequest=2) it.</sub>

sveitser committed:
ci: Use self hosted runner with docker

- Shorten tmp dir path: otherwise tests fail because a path becomes
  too long (>107 chars) for an IPC socket.

sveitser committed:
ci: create tmp dir

sveitser committed:
ci: print redis logs

sveitser committed:
ci: remove redis logs

sveitser committed:
ci: redis, use docker network alias

sveitser committed:
ci: fix yaml

nomaxg committed:
surpress G114 timeout issue

nomaxg committed:
try using absolute go.sum path
